library/core/src/ffi/c_str.rs

```diff
@@ -742,6 +740,8 @@ impl AsRef<CStr> for CStr {
 /// The pointer must point to a valid buffer that contains a NUL terminator. The NUL must be
 /// located within `isize::MAX` from `ptr`.
 #[inline]
+#[rustc_const_stable(feature = "const_cstr_from_ptr", since = "CURRENT_RUSTC_VERSION")]
```

Comment:
Assuming the attribute on a nonpublic function allows using `const_strlen` everywhere without adding `const_eval_select` each time?

Reply:
I would prefer not to add such misuses of stability attributes. It makes it harder to review internal changes, because an important part of T-libs reviews when you _aren't_ reviewing an addition API is making sure that you are not, in fact, adding API. This weakens such a defense and makes it easier for a reviewer to get confused.

Reply:
In what way is this a misuse? The lang and libs teams FCPd having a `const_strlen` available in some form for internal use, currently via `const_eval_select` but that is an implementation detail. Marking it as "stable" (but of course nonpublic) seems to accurately reflect that status, as opposed to leaking the `const_eval_select` implementation detail around everywhere it is used.

Reply:
Right @tgross35, I believe this is const stability working as intended. The attribute enforces that future changes to this function do not introduce any unapproved implementation, because this private function now participates in the implementation of some public stable const function and the use of any new weird const internals here would become a stable commitment. Flagging the presence of a stable commitment is the point of a stability attribute.

@RalfJung endorsed the same thing recently in https://github.com/rust-lang/rust/pull/124941#discussion_r1664676739.

Reply:
It would probably be okay to rename `const_strlen` to just `strlen` at some point to make it more clear that's just our implementation of `strlen` now (avoiding confusion like https://github.com/rust-lang/rust/pull/127444).

Reply:
I believe it should have a comment disclaiming its status, then, rather than passing unremarked, because this strikes me as a nonintuitive combination of attributes when placed in a public module.

Or it could be marked `#[unstable]`, as the other function cited is.

Reply:
I have added `unstable` in https://github.com/rust-lang/rust/pull/127433/commits/7f1518bddd76ad8e3a743278fd983c35d11b2411.

Reply:
Thank you. "Fix this one attribute" or "oops this stabilization missed a spot" or "make this thing public that obviously was intended to be public... right?" are all fairly common PRs.